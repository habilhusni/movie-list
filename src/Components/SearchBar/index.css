.ant-input-wrapper {
  max-width: 500px;
  margin-top: 50px !important;
  margin-left: auto !important;
  margin-right: auto !important;
}
