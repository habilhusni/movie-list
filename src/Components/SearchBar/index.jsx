import React from "react";
import { Input } from "antd";
import "./index.css";

const { Search } = Input;

const handleSearchBar = (prps, keyword) => {
  if (prps) {
    prps.searchMovies(keyword);
  }
};

const SearchBar = (props) => (
  <Search
    placeholder="Search movies by title"
    enterButton="Search"
    size="large"
    onSearch={(value) => handleSearchBar(props, value)}
  />
);

export default SearchBar;
