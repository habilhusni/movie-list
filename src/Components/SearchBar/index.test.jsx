import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJSON from "enzyme-to-json";

import SearchBar from "./index.jsx";

Enzyme.configure({ adapter: new Adapter() });

describe("Test Snapshot and Functional", () => {
  let component;
  let props = {
    getMovies: () => {},
  };

  beforeEach(() => {
    component = shallow(<SearchBar {...props} />);
  });

  it("Testing SearchBar Snapshot", () => {
    expect(toJSON(component)).toMatchSnapshot();
  });
});
