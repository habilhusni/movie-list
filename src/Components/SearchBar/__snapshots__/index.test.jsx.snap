// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Snapshot and Functional Testing SearchBar Snapshot 1`] = `
<Search
  enterButton="Search"
  onSearch={[Function]}
  placeholder="Search movies by title"
  size="large"
/>
`;
